torch/onnx/_internal/exporter.py

```diff
@@ -61,13 +69,46 @@ def __init__(
         *,
         opset_version: Optional[int] = None,
         dynamic_shapes: Optional[bool] = None,
+        op_level_debug: Optional[bool] = None,
         logger: Optional[logging.Logger] = None,
     ):
         self.opset_version = opset_version
         self.dynamic_shapes = dynamic_shapes
+        self.op_level_debug = op_level_debug
         self.logger = logger
 
 
+class ResolvedExportOptions:
+    @_beartype.beartype
+    def __init__(self, options: Optional[ExportOptions]):
+        if options is None:
+            options = ExportOptions()
+
+        T = TypeVar("T")
+
+        @_beartype.beartype
+        def resolve(value: Optional[T], fallback: Union[T, Callable[[], T]]) -> T:
+            if value is not None:
+                return value
+            if callable(fallback):
+                return fallback()
+            return fallback
```

Comment:
I would move the function out to module namespace unless if it's closing over local variables.

https://google.github.io/styleguide/pyguide.html#264-decision